packages/devtools_app/lib/src/charts/flame_chart_canvas.dart

```diff
@@ -218,7 +247,7 @@ abstract class FlameChartCanvas<T> extends FlameChart {
       rowIndexForY(rect.bottom) + 1,
       rows.length - 1,
     );
-    for (int i = startRow; i < endRow; i++) {
+    for (int i = startRow; i <= endRow; i++) {
```

Comment:
why is this <= endRow?. Add a comment as that is surprising.

Reply:
This is so that we ensure the end row is painted. Changed the name to `endRowInclusive` so that it is self-documenting. Also modified `rowIndexForY(rect.bottom) + 1` to `rowIndexForY(rect.bottom)` so that we are not painting an extra row below what is visible.